tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_Shape.cpp

```diff
@@ -0,0 +1,442 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <complex>
+#include <cstddef>
+#include <functional>
+#include <memory>
+#include <random>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "ApparentHorizons/SpherepackIterator.hpp"
+#include "ApparentHorizons/Tags.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/CoordinateMaps/TimeDependent/Shape.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Helpers/Domain/CoordinateMaps/TestMapHelpers.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+namespace domain {
+namespace {
+
+const std::complex<double> imag(0, 1);
+
+std::complex<double> Y00(double /*theta*/, double /*phi*/) {
+  return 0.5 * sqrt(1. / M_PI);
+}
+
+std::complex<double> Y10(double theta, double /*phi*/) {
+  return 0.5 * sqrt(3. / M_PI) * cos(theta);
+}
+
+std::complex<double> Y11(double theta, double phi) {
+  return -0.5 * sqrt(3. / (2. * M_PI)) * sin(theta) *
+         std::exp<double>(imag * phi);
+}
+
+std::complex<double> Y20(double theta, double /*phi*/) {
+  return 0.25 * sqrt(5. / M_PI) * (3. * cos(theta) * cos(theta) - 1.);
+}
+
+std::complex<double> Y21(double theta, double phi) {
+  return -0.5 * sqrt(15. / (2. * M_PI)) * sin(theta) * cos(theta) *
+         std::exp<double>(imag * phi);
+}
+
+std::complex<double> Y22(double theta, double phi) {
+  return 0.25 * sqrt(15. / (2. * M_PI)) * sin(theta) * sin(theta) *
+         std::exp<double>(2. * imag * phi);
+}
+
+std::complex<double> Y30(double theta, double /*phi*/) {
+  return 0.25 * sqrt(7. / M_PI) * (5 * pow(cos(theta), 3) - 3 * cos(theta));
+}
+
+std::complex<double> Y31(double theta, double phi) {
+  return -0.125 * sqrt(21. / M_PI) * sin(theta) *
+         (5 * cos(theta) * cos(theta) - 1) * std::exp<double>(imag * phi);
+}
+
+std::complex<double> Y32(double theta, double phi) {
+  return 0.25 * sqrt(105. / (2. * M_PI)) * sin(theta) * sin(theta) *
+         cos(theta) * std::exp<double>(2. * imag * phi);
+}
+
+std::complex<double> Y33(double theta, double phi) {
+  return -0.125 * sqrt(35. / M_PI) * pow(sin(theta), 3) *
+         std::exp<double>(3. * imag * phi);
+}
+
+// vector of spherical harmonics, e.g. Y21 is harmonics[2][1]
+using SphericalHarmonic = std::function<std::complex<double>(double, double)>;
+std::vector<std::vector<SphericalHarmonic>> harmonics = {
+    {Y00}, {Y10, Y11}, {Y20, Y21, Y22}, {Y30, Y31, Y32, Y33}};
+
+auto c_coef = [](size_t l, size_t m) {
```

Comment:
This can just be a normal function, no? Please add a comment to describe it a bit